Building HPX with support for PAPI fails

## Expected Behavior

Build succeeds

## Actual Behavior

Builds fails. Error message:

```
/tmp/hello_hpx-fetch_content/RelWithDebInfo-papi/build/_deps/hpx-src/components/performance_counters/papi/src/papi_startup.cpp: In function ‘void hpx::performance_counters::papi::find_threads(std::set<std::__cxx11::basic_string<char> >&, hpx::performance_counters::discover_counters_mode)’:
/tmp/hello_hpx-fetch_content/RelWithDebInfo-papi/build/_deps/hpx-src/components/performance_counters/papi/src/papi_startup.cpp:110:51: error: invalid use of incomplete type ‘class hpx::runtime’
         hpx::util::thread_mapper& tm = get_runtime().get_thread_mapper();
```

[build_output.txt](https://github.com/STEllAR-GROUP/hpx/files/4779144/build_output.txt)

## Steps to Reproduce the Problem

  1. Build HPX with support for PAPI. My script used this command to configure HPX: `cmake -DCMAKE_BUILD_TYPE=/tmp/hello_hpx-fetch_content/RelWithDebInfo-papi -DHELLO_HPX_REPOSITORY_CACHE:PATH=/home/kor/development/repository papi -DHPX_WITH_PAPI:BOOL=ON -HPX_WITH_MALLOC:STRING=tcmalloc -HPX_WITH_HWLOC:BOOL=ON -S /home/kor/development/project/hello_hpx -B /tmp/hello_hpx-fetch_content/RelWithDebInfo-papi/build`
  1. Build HPX
  1. Wait for error message. Occurs at 61% of the build in my case.

## Specifications

  - HPX Version: master of last Friday (commit 47d7bebd27a)
  - Platform (compiler, OS): gcc-7, Kubuntu 18.04

--- a/components/containers/partitioned_vector/include/hpx/components/containers/partitioned_vector/partitioned_vector_component_decl.hpp
+++ b/components/containers/partitioned_vector/include/hpx/components/containers/partitioned_vector/partitioned_vector_component_decl.hpp
@@ -333,7 +333,11 @@ namespace hpx
             > base_type;
 
     public:
+#if defined(HPX_CLANG_VERSION) && HPX_CLANG_VERSION < 60000
+        partitioned_vector_partition() {};
+#else
         partitioned_vector_partition() = default;
+#endif
 
         explicit partitioned_vector_partition(id_type const& gid);
 

--- a/components/performance_counters/papi/src/papi_startup.cpp
+++ b/components/performance_counters/papi/src/papi_startup.cpp
@@ -12,14 +12,13 @@
 #include <hpx/components/performance_counters/papi/server/papi.hpp>
 #include <hpx/components/performance_counters/papi/util/papi.hpp>
 #include <hpx/modules/errors.hpp>
+#include <hpx/modules/runtime_local.hpp>
 #include <hpx/performance_counters/counter_creators.hpp>
 #include <hpx/performance_counters/manage_counter_type.hpp>
-#include <hpx/runtime.hpp>
 #include <hpx/runtime/components/component_commandline.hpp>
 #include <hpx/runtime/components/component_factory_base.hpp>
 #include <hpx/runtime/components/component_startup_shutdown.hpp>
 #include <hpx/runtime/components/server/create_component.hpp>
-#include <hpx/runtime_local/startup_function.hpp>
 #include <hpx/functional/bind_back.hpp>
 #include <hpx/runtime_local/thread_mapper.hpp>
 

--- a/components/performance_counters/papi/src/server/papi.cpp
+++ b/components/performance_counters/papi/src/server/papi.cpp
@@ -12,11 +12,11 @@
 #include <hpx/components/performance_counters/papi/server/papi.hpp>
 #include <hpx/components/performance_counters/papi/util/papi.hpp>
 #include <hpx/modules/errors.hpp>
-#include <hpx/runtime.hpp>
+#include <hpx/modules/runtime_local.hpp>
+#include <hpx/modules/timing.hpp>
 #include <hpx/runtime/actions/continuation.hpp>
 #include <hpx/runtime/components/derived_component_factory.hpp>
 #include <hpx/runtime/components/server/component.hpp>
-#include <hpx/modules/timing.hpp>
 #include <hpx/runtime_local/thread_mapper.hpp>
 
 #include <boost/version.hpp>

--- a/components/performance_counters/papi/src/util/papi.cpp
+++ b/components/performance_counters/papi/src/util/papi.cpp
@@ -9,11 +9,10 @@
 #if defined(HPX_HAVE_PAPI)
 
 #include <hpx/config/asio.hpp>
+#include <hpx/modules/command_line_handling.hpp>
 #include <hpx/modules/errors.hpp>
-#include <hpx/runtime.hpp>
 #include <hpx/modules/format.hpp>
-#include <hpx/command_line_handling/parse_command_line.hpp>
-#include <hpx/runtime_local/thread_mapper.hpp>
+#include <hpx/modules/runtime_local.hpp>
 #include <hpx/components/performance_counters/papi/util/papi.hpp>
 
 #include <boost/asio/ip/host_name.hpp>

